tests/parsers/test_google_decimeter.py

```diff
@@ -0,0 +1,823 @@
+
+    """
+
+    pd_df_xl = pd.read_csv(derived_path_xl)
+    derived_xl = google_decimeter.AndroidDerived2021(derived_path_xl,
+                               remove_timing_outliers=False)
+
+    pd_svid_groups = []
+    for _, group in pd_df_xl.groupby("millisSinceGpsEpoch"):
+        pd_svid_groups.append(group["svid"].tolist())
+    pd_svid_groups.pop(0)
+
+    navdata_svid_groups = []
+    for _, _, group in derived_xl.loop_time("gps_millis"):
+        navdata_svid_groups.append(group["sv_id"].astype(int).tolist())
+
+    assert len(pd_svid_groups) == len(navdata_svid_groups)
+
+    for pd_ids, navdata_ids in zip(pd_svid_groups,navdata_svid_groups):
+        assert pd_ids == navdata_ids
+
+
+def test_shape_update(derived):
+    """Test that shape gets updated after adding a row.
+
+    Parameters
+    ----------
+    derived : pytest.fixture
+        Instance of AndroidDerived2021 for testing
+    """
+    old_shape = derived.shape
+    derived["new_row"] = np.ones((old_shape[1]))
+    new_shape = derived.shape
+
+    # should still have the same number of columns (timesteps)
+    np.testing.assert_equal(old_shape[1], new_shape[1])
+    # should have added one new row
+    np.testing.assert_equal(old_shape[0] + 1, new_shape[0])
+
+@pytest.fixture(name="android_gtruth_path")
+def fixture_gtruth_path(root_path):
+    """Filepath of Android Ground Truth data
+
+    Returns
+    -------
+    gtruth_path : string
+        Location for text log file with Android Ground Truth measurements
+
+    Notes
+    -----
+    Test data is a subset of the Android Ground Truth Dataset [4]_,
+    particularly the train/2020-05-14-US-MTV-1/Pixel4 trace. The dataset
+    was retrieved from
+    https://www.kaggle.com/c/google-smartphone-decimeter-challenge/data
+
+    References
+    ----------
+    .. [4] Fu, Guoyu Michael, Mohammed Khider, and Frank van Diggelen.
+        "Android Raw GNSS Measurement Datasets for Precise Positioning."
+        Proceedings of the 33rd International Technical Meeting of the
+        Satellite Division of The Institute of Navigation (ION GNSS+
+        2020). 2020.
+    """
+    gtruth_path = os.path.join(root_path, 'Pixel4_ground_truth.csv')
+    return gtruth_path
+
+@pytest.fixture(name="gtruth")
+def fixture_load_gtruth(android_gtruth_path):
+    """Load instance of AndroidGroundTruth2021
+
+    Parameters
+    ----------
+    gtruth_path : pytest.fixture
+        String with location of Android ground truth file
+
+    Returns
+    -------
+    gtruth : AndroidGroundTruth2021
+        Instance of AndroidGroundTruth2021 for testing
+    """
+    gtruth = google_decimeter.AndroidGroundTruth2021(android_gtruth_path)
+
+    return gtruth
+
+def test_android_gtruth(gtruth):
+    """Test AndroidGroundTruth initialization
+
+    Parameters
+    ----------
+    gtruth : AndroidGroundTruth2021
+        Instance of AndroidGroundTruth2021 for testing
+
+    """
+
+    isinstance(gtruth, NavData)
```

Comment:
Similarly, an assert statement may be useful here for clarity.

Reply:
Updated 13 places throughout the repo